scipy/interpolate/_rgi.py

```diff
@@ -621,6 +622,14 @@ def interpn(points, values, xi, method="linear", bounds_error=True,
                          "2-dimensional input data")
     if not bounds_error and fill_value is None and method == "splinef2d":
         raise ValueError("The method splinef2d does not support extrapolation.")
+    
+    if method == "pchip" and np.iscomplexobj(values):
+        msg = ("Passing complex-dtyped `values` with `method='pchip'` "
+               "is deprecated and will raise an error in SciPy 1.15.0. "
+               "This interpolation method only works with real values. "
+               "If you are trying to use the real components, use `np.real` on "     
+               "the array before passing to `interpn`.")
+        warnings.warn(msg, DeprecationWarning, stacklevel=2)
```

Comment:
This warning should originate in RegularGridInterpolator since `interpn` is just a convenience wrapper. 
In fact, a DeprecationWarning from PchipInterpolator may be enough? Up to you.